main.php

```diff
@@ -2,153 +2,166 @@
+				echo '<p>' . esc_html__( 'Child theme does not have the <strong>Template Version</strong> tag in style.css.', 'theme-check' ) . '</p>';
 			} else {
-				echo ( $data['Template Version'] < $parent_data['Version'] ) ? '<p>' . sprintf(__('Child theme is only tested up to version %1$s of %2$s breakage may occur! %3$s installed version is %4$s', 'theme-check'), $data['Template Version'], $parent_data['Title'], $parent_data['Title'], $parent_data['Version'] ) . '</p>' : '';
+				echo ( $data['Template Version'] < $parent_data['Version'] ) ? '<p>' . sprintf( esc_html__( 'Child theme is only tested up to version %1$s of %2$s breakage may occur! %3$s installed version is %4$s', 'theme-check' ), $data['Template Version'], $parent_data['Title'], $parent_data['Title'], $parent_data['Version'] ) . '</p>' : '';
 			}
-		 }
+		}
 		echo '</div><!-- .theme-info-->';
 
 		$plugins = get_plugins( '/theme-check' );
 		$version = explode( '.', $plugins['theme-check.php']['Version'] );
 		echo '<p>' . sprintf(
-			__(' Running %1$s tests against %2$s using Guidelines Version: %3$s Plugin revision: %4$s', 'theme-check'),
+			esc_html__( ' Running %1$s tests against %2$s using Guidelines Version: %3$s Plugin revision: %4$s', 'theme-check' ),
 			'<strong>' . $checkcount . '</strong>',
-			'<strong>' . $data[ 'Title' ] . '</strong>',
+			'<strong>' . $data['Title'] . '</strong>',
 			'<strong>' . $version[0] . '</strong>',
 			'<strong>' . $version[1] . '</strong>'
 		) . '</p>';
 		$results = display_themechecks();
-		if ( !$success ) {
-			echo '<h2>' . sprintf(__('One or more errors were found for %1$s.', 'theme-check'), $data[ 'Title' ] ) . '</h2>';
+		if ( ! $success ) {
+			echo '<h2>' . sprintf( esc_html__( 'One or more errors were found for %1$s.', 'theme-check' ), $data['Title'] ) . '</h2>';
 		} else {
-			echo '<h2>' . sprintf(__('%1$s passed the tests', 'theme-check'), $data[ 'Title' ] ) . '</h2>';
+			echo '<h2>' . sprintf( __( '%1$s passed the tests', 'theme-check' ), $data['Title'] ) . '</h2>';
 			tc_success();
 		}
-		if ( !defined( 'WP_DEBUG' ) || WP_DEBUG == false ) echo '<div class="updated"><span class="tc-fail">' . __('WARNING','theme-check') . '</span> ' . __( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
+		if ( ! defined( 'WP_DEBUG' ) || WP_DEBUG === false ) {
+			echo '<div class="updated"><span class="tc-fail">' . esc_html__( 'WARNING', 'theme-check' ) . '</span> ' . esc_html__( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
+		}
 		echo '<div class="tc-box">';
 		echo '<ul class="tc-result">';
-		echo $results;
+		echo esc_html( $results );
```

Comment:
Using esc_html here breaks the output of the results which makes use of lists etc.